sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
@weiqingy , even if you add a wrapper to try to support different stream handler factory. It is not a good idea to filter with only hdfs. `FsUrlStreamHandlerFactory` by default supports different fs implementation, it is not necessary to check deliberately by the upstream code.